tests/test_inprocess.py

```diff
@@ -0,0 +1,160 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import subprocess
+from typing import TYPE_CHECKING
+
+import pytest
+
+if TYPE_CHECKING:
+    from ansys.mapdl.core import Mapdl
+
+from conftest import ON_LOCAL
+
+
+class MapdlInProcessRunner:
+    def __init__(self, wdir: str, exec_path: str | None = None) -> None:
+        self.wdir = str(wdir)
+        self.completed_process: subprocess.CompletedProcess[bytes] | None = None
+        self._exec_path: str | None = exec_path
+
+    @property
+    def exec_path(self) -> str:
+        if self._exec_path is None:
+
+            self._exec_path = os.getenv("PYMAPDL_MAPDL_EXEC")
+            if self._exec_path is None:
+                from ansys.tools.path import get_mapdl_path
+
+                version: float | None = (
+                    float(os.getenv("PYMAPDL_MAPDL_VERSION", 0)) or None
+                )
+                self._exec_path = get_mapdl_path(version=version)
+
+        if self._exec_path is None:
+            raise ValueError(
+                "MAPDL executable path is not set. Set the PYMAPDL_MAPDL_EXEC "
+                "environment variable or use get_mapdl_path() to find it."
+            )
+
+        return self._exec_path
+
+    @property
+    def stdout(self) -> str | None:
+        if self.completed_process:
+            return self.completed_process.stdout.decode()
+
+    @property
+    def stderr(self) -> str | None:
+        if self.completed_process:
+            return self.completed_process.stderr.decode()
+
+    def run(self, cmds: str) -> str:
+        """Run commands in MAPDL on batch mode."""
+        with open(os.path.join(self.wdir, "input.mac"), "w") as f:
+            f.write(cmds)
+
+        try:
+            args = [
+                self.exec_path,
+                "-b",
+                "-i",
+                "input.mac",
+                "-o",
+                "out.out",
+                os.getenv("PYMAPDL_ADDITIONAL_SWITCHES", ""),
+            ]
+
+            self.completed_process = subprocess.run(
+                args=args,
+                cwd=self.wdir,
+                check=True,
+                capture_output=True,
+                # it does not support shell=True, because it does not
```

Comment:
worth investigating, maybe add a TODO for later?

Reply:
done